Several errors compiling brand new project to iOS

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.2 ec39e38
**OS:** macOS Big Sur 11.5.1

**What did you do?**

- v new test
- cd test
- v -os ios .

*Extra note*:

I asked for #help in Discord and thanks to @spytheman and @spaceface777 I was able to compile after editing **thirdparty/ios/ios.m** to the following:

```
void WrappedNSLog(const char *message,...) {  
    va_list args;  
    va_start(args, message);  
    va_end(args);  
}  
```
**What did you expect to see?** 

A compiler "test" executable for iOS
 
**What did you see instead?**
```
                     ^
/tmp/v/test.7630996033059445783.tmp.c:2388:22: error: conflicting types for 'strconv__is_space'
VV_LOCAL_SYMBOL bool strconv__is_space(byte x) {
                     ^
/tmp/v/test.7630996033059445783.tmp.c:1235:22: note: previous declaration is here
VV_LOCAL_SYMBOL bool strconv__is_space(byte x);
                     ^
/tmp/v/test.7630996033059445783.tmp.c:2392:22: error: conflicting types for 'strconv__is_exp'
VV_LOCAL_SYMBOL bool strconv__is_exp(byte x) {
                     ^
/tmp/v/test.7630996033059445783.tmp.c:1236:22: note: previous declaration is here
VV_LOCAL_SYMBOL bool strconv__is_exp(byte x);
...
```

--- a/vlib/v/builder/cc.v
+++ b/vlib/v/builder/cc.v
@@ -523,17 +523,7 @@ pub fn (mut v Builder) cc() {
 		// try to compile with the choosen compiler
 		// if compilation fails, retry again with another
 		mut ccompiler := v.pref.ccompiler
-		if v.pref.os == .ios {
-			ios_sdk := if v.pref.is_ios_simulator { 'iphonesimulator' } else { 'iphoneos' }
-			ios_sdk_path_res := os.execute_or_exit('xcrun --sdk $ios_sdk --show-sdk-path')
-			mut isysroot := ios_sdk_path_res.output.replace('\n', '')
-			arch := if v.pref.is_ios_simulator {
-				'-arch x86_64'
-			} else {
-				'-arch armv7 -arch armv7s -arch arm64'
-			}
-			ccompiler = 'xcrun --sdk iphoneos clang -isysroot $isysroot $arch'
-		} else if v.pref.os == .wasm32 {
+		if v.pref.os == .wasm32 {
 			ccompiler = 'clang'
 		}
 		v.setup_ccompiler_options(ccompiler)
@@ -619,7 +609,7 @@ pub fn (mut v Builder) cc() {
 				}
 				if v.pref.retry_compilation {
 					tcc_output = res
-					v.pref.ccompiler = pref.default_c_compiler()
+					v.pref.default_c_compiler()
 					if v.pref.is_verbose {
 						eprintln('Compilation with tcc failed. Retrying with $v.pref.ccompiler ...')
 					}

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -652,6 +652,7 @@ pub fn (mut g Gen) init() {
 				g.cheaders.writeln('#include <stddef.h>')
 			} else {
 				g.cheaders.writeln(get_guarded_include_text('<inttypes.h>', 'The C compiler can not find <inttypes.h>. Please install build-essentials')) // int64_t etc
+				g.cheaders.writeln(get_guarded_include_text('<stdbool.h>', 'The C compiler can not find <stdbool.h>. Please install build-essentials')) // bool, true, false
 				g.cheaders.writeln(get_guarded_include_text('<stddef.h>', 'The C compiler can not find <stddef.h>. Please install build-essentials')) // size_t, ptrdiff_t
 			}
 		}

--- a/vlib/v/pref/default.v
+++ b/vlib/v/pref/default.v
@@ -92,7 +92,7 @@ pub fn (mut p Preferences) fill_with_defaults() {
 	//
 	p.try_to_use_tcc_by_default()
 	if p.ccompiler == '' {
-		p.ccompiler = default_c_compiler()
+		p.default_c_compiler()
 	}
 	p.find_cc_if_cross_compiling()
 	p.ccompiler_type = cc_from_string(p.ccompiler)
@@ -203,16 +203,32 @@ pub fn default_tcc_compiler() string {
 	return ''
 }
 
-pub fn default_c_compiler() string {
+pub fn (mut p Preferences) default_c_compiler() {
 	// fast_clang := '/usr/local/Cellar/llvm/8.0.0/bin/clang'
 	// if os.exists(fast_clang) {
 	// return fast_clang
 	// }
 	// TODO fix $if after 'string'
 	$if windows {
-		return 'gcc'
+		p.ccompiler = 'gcc'
+		return
 	}
-	return 'cc'
+	if p.os == .ios {
+		$if !ios {
+			ios_sdk := if p.is_ios_simulator { 'iphonesimulator' } else { 'iphoneos' }
+			ios_sdk_path_res := os.execute_or_exit('xcrun --sdk $ios_sdk --show-sdk-path')
+			mut isysroot := ios_sdk_path_res.output.replace('\n', '')
+			arch := if p.is_ios_simulator {
+				'-arch x86_64'
+			} else {
+				'-arch armv7 -arch armv7s -arch arm64'
+			}
+			p.ccompiler = 'xcrun --sdk iphoneos clang -isysroot $isysroot $arch'
+			return
+		}
+	}
+	p.ccompiler = 'cc'
+	return
 }
 
 pub fn vexe_path() string {

